src/util/Util.cpp

```diff
@@ -165,3 +166,17 @@ gboolean Util::paintBackgroundWhite(GtkWidget* widget, cairo_t* cr, void* unused
 	return false;
 }
 
+char* Util::getCoordinateString(double xVal, double yVal)
+{
+	char tmpX[G_ASCII_DTOSTR_BUF_SIZE];
+	g_ascii_formatd(tmpX, G_ASCII_DTOSTR_BUF_SIZE, Util::getFormatdPrecisionString(), xVal);
+	char tmpY[G_ASCII_DTOSTR_BUF_SIZE];
+	g_ascii_formatd(tmpY, G_ASCII_DTOSTR_BUF_SIZE, Util::getFormatdPrecisionString(), yVal);
+
+	return g_strdup_printf("%s %s", tmpX, tmpY);
```

Comment:
g_strdup_printf allocates memory that needs to be freed.  Our function should also mention this in a doxygen type comment block ( some IDE's will pop this up on mouseover).

**Better yet,**   use g_snprintf () instead and require the caller to allocate and pass us a buffer to use
 ( so that it becomes `getCoordinateString(gchar*buff, gulong buffLen, double xVal, double yVal) `    ) 
I see two benefits to using g_snprintf:  1) no hidden allocations, 2) Can re-use the buffer when calling this function repeatedly in a loop ( _although I see the original code misses this optimization.. maybe the compiler did it ;)_  )

One could also consider using a c++ object that free's in the destructor... but I think that's overkill here.

Reply:
I plan to use `g_snprintf` because of those benefits that you mentioned.